omod/src/main/webapp/resources/scripts/jquery-ui/js/openmrsSearch.js

```diff
@@ -546,26 +546,38 @@ function OpenmrsSearch(div, showIncludeVoided, searchHandler, selectionHandler,
             });
 
             // Browser back button support, for preserve data
-            if (lastSearchParams !== null) {
-                $j('#inputNode').val(lastSearchParams.lastSearchText);
-                if (lastSearchParams.includeVoided == 1) {
+            var lastSearchText = decodeURIComponent((new RegExp('[?|&]' + 'lastSearchText' 
+            		+ '=' + '([^&;]+?)(&|#|;|$)').exec(location.search)||[,""])[1].replace(/\+/g, '%20')) || null;
+            var includeVoided = decodeURIComponent((new RegExp('[?|&]' + 'includeVoided' 
+            		+ '=' + '([^&;]+?)(&|#|;|$)').exec(location.search)||[,""])[1].replace(/\+/g, '%20')) || null;
+            var includeVerbose = decodeURIComponent((new RegExp('[?|&]' + 'includeVerbose' 
+            		+ '=' + '([^&;]+?)(&|#|;|$)').exec(location.search)||[,""])[1].replace(/\+/g, '%20')) || null;
+            
+            //if we have lastSearchText then set input text, move cursor to the end and trigger a keyup event to load the search
+            if (lastSearchText !== null) {
+            	
+                $j('#inputNode').val(lastSearchText);
```

Comment:
Instead of this, you need to assign the value of lastSearchText to options.searchPhrase instead, you will need to make the widget support showVerbose and includeVoided as widget options too.